md/barostat/utils.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+import networkx as nx
+from scipy.spatial.distance import pdist
+from typing import List
+from timemachine.lib.potentials import HarmonicBond
+
+
+def compute_box_volume(box: np.ndarray) -> float:
+    assert box.shape == (3, 3)
+    return np.linalg.det(box)
+
+
+def compute_box_center(box: np.ndarray) -> np.ndarray:
+    # assume axis-aligned box (nothing off diagonal)
+    assert box.shape == (3, 3)
+    assert np.linalg.norm(box - np.diag(np.diag(box))) == 0
+
+    center = np.sum(box / 2, axis=0)
+
+    assert center.shape == (3,)
+
+    return center
+
+
+def get_group_indices(harmonic_bond_potential: HarmonicBond) -> List[np.array]:
+    # read off topology from harmonic bond indices
+    # NOTE: this assumes all bonds are represented by harmonic bond force
+    bond_list = list(map(tuple, harmonic_bond_potential.get_idxs()))
+    # TODO: if we add HBond constraints, be sure to add these to the bond_list!
+    alchemical_topology = nx.Graph(bond_list)
+    connected_components = list(map(list, nx.algorithms.connected_components(alchemical_topology)))
+
+    return connected_components
+
+
+def merge_big_groups(group_indices: List[np.array]) -> List[np.array]:
```

Comment:
is this still being used?

Reply:
Ahh, good catch -- this had been used in an earlier version where I wanted to have the barostat displace the protein:ligand complex as a single unit (rather than displacing the protein and ligand separately). Examples don't currently exercise this option, and I don't currently see a compelling reason to do it one way or the other, so I removed this function in https://github.com/proteneer/timemachine/pull/433/commits/03952ce62b88fecf1a7c43408413db7ea9796175 .